src/yaml/mapping.cr

```diff
@@ -181,6 +181,14 @@ module YAML
             {% end %}
           end
         {% end %}
+        {% if extra %}
+          _{{extra.id}} = @{{extra.id}}
```

Comment:
if `@{{extra.id}}` exists, then it is not nillable so you could call `@{{extra.id}}.to_yaml(...` directly. For memory footprint I would suggest to let it be lazy initialized / nilable. But for the use case, if the user expect to have some extras, maybe is good enough to eagerly create a and instance for `@{{extra.id}}`.